rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -93,6 +94,41 @@
 MIN_LOGS_PROCESSED_TO_SLEEP = 1000
 
 
+def decoding_wrapper(
+        msg_aggregator: 'MessagesAggregator | None',
+        transaction_for_logs: EvmTransaction,
```

Comment:
Why are you passing this? The name `transaction_for_logs` implies it's only for logs. But then why the entire class? Even if it's just for logs give it a proper name. Just call it `transaction`.

```suggestion
        transaction_for_logs: EvmTransaction,
```

Why not just pass the transaction hash and chain id? Since this passes a lot more info too. Though this may be just a nitpick.

Reply:
the reason to not use `transaction` as name is because if it is used as kwarg in the `func` being wrapped it will appear twice and python complains